darjutak-2018-november-sip-minutes.md

```diff
@@ -0,0 +1,122 @@
+3. **Tooling to ease upgrades.** There should be a rewriting tool for automatic migrations and changes in the current Scala tooling. This tool should accommodate most of the needed changes.
+4. **Shared standard library.** Scala 2.14 and Scala 3.0 should share the same Scala standard library.
+
+In addition, the Scala community has a culture of upgrading the ecosystem on every major version of Scala through the community build. We  recommend to leverage the same mechanism to evaluate Scala upgrades with the migration to Scala 3.
+
+### How will the transition to Scala 3 affect users of Scala 2 macros and reflection?
+
+The dependency of Scala 2 macros and reflection on internal implementation details of the Scala 2 compiler mean that significant change is inevitable if Scala is to evolve.
+We recognize that important parts of the Scala ecosystem have made essential use of the Scala 2 facilities and that it is vital that as many as possible of these use cases be accommodated in Scala 3 in some form or another. This will be disruptive, but we hope to mitigate the disruption by providing facilities which make the more straightforward and important scenarios simpler whilst leaving others still possible.
+Our direction is still evolving, however we believe that replacing the current excessively general and expressive macro system with a suite of less powerful but complementary tools is the way forward.
+Currently we are exploring options which range from improved support for type level programming in the language itself (eg. specialized inline, match types, stable definitions, GADT improvements); intensifying certain features currently supported by macros (eg. by-name implicits, generic programming primitives); through to less general forms of metaprogramming (quote/splice and staging) and portable reflection via TASTy (which we recommend to support in both Scala 2/3 and via compiler-independent libraries and tools; [see also](https://github.com/scalacenter/advisoryboard/pull/40)). We recommend that most current uses of Scala macros and reflection can be accommodated by some combination of these tools.
+For more about the project progress, please see https://github.com/lampepfl/dotty/issues/5489
+
+### How do we plan to address language experimentation?
+We acknowledge that language experimentation is necessary for improving the language. We also believe it requires a different vehicle than stable Scala releases. We don’t have a concrete solution for now, but we’re working on one.
+
+### Other “documents” created during the meetings:
+
+https://github.com/lampepfl/dotty/issues/5372
+
+https://github.com/lampepfl/dotty/issues/5378
+
+https://github.com/lampepfl/dotty/issues/5379
+
+https://github.com/lampepfl/dotty/issues/5381
+
+
+Features work progress overview:
+
+https://docs.google.com/spreadsheets/d/1GWJUo0U3JbBtrfg5vqgb6H5S6wlU5HnTxebLcHwD1zw/edit?usp=sharing 
```

Comment:
The list of what features are "core" is probably the single thing in this whole document of the greatest specific interest to readers; I think it might be good to include that list in the minutes directly, as well, as providing the link to the full spreadsheet.

Reply:
totally agreed!